IDEAS/Examples/IBPSA/SingleZoneResidentialHydronicHeatPump.mo

```diff
@@ -0,0 +1,709 @@
+  Utilities.IO.SignalExchange.Read reaQHeaPumEva(
+    description="Heat pump thermal power exchanged in the evaporator",
+    KPIs=IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.None,
+    y(unit="W"))
+    "Block for reading the heat pump thermal power exchanged in the evaporator"
+    annotation (Placement(transformation(extent={{160,0},{180,20}})));
+  Utilities.IO.SignalExchange.Read reaQHeaPumCon(
+    description="Heat pump thermal power exchanged in the condenser",
+    KPIs=IDEAS.Utilities.IO.SignalExchange.SignalTypes.SignalsForKPIs.None,
+    y(unit="W"))
+    "Block for reading the heat pump thermal power exchanged in the condenser"
+    annotation (Placement(transformation(extent={{100,0},{80,20}})));
+  Utilities.IO.SignalExchange.Read reaTSup(description="Supply water temperature to emission system",
+      y(unit="K")) "Read supply water temperature to emission system"
+    annotation (Placement(transformation(extent={{100,70},{120,90}})));
+  Utilities.IO.SignalExchange.Read reaTRet(description="Return water temperature from emission system",
+      y(unit="K")) "Read return water temperature from emission system"
+    annotation (Placement(transformation(extent={{100,-60},{120,-40}})));
+  Utilities.IO.SignalExchange.Read reaCOP(description="Heat pump COP", y(unit=
+          "1")) "Read heat pump COP"
+    annotation (Placement(transformation(extent={{200,-60},{220,-40}})));
+  Modelica.Blocks.Sources.RealExpression heaPumCOP(y=heaPum.com.COP)
+    "Substracts heat pump COP"
+    annotation (Placement(transformation(extent={{160,-60},{180,-40}})));
+  Fluid.Sources.OutsideAir outAir(redeclare package Medium = MediumAir,
+                                  nPorts=2) "Outside air"
+    annotation (Placement(transformation(extent={{260,0},{240,20}})));
+  Modelica.Blocks.Math.RealToInteger realToInteger2
+    annotation (Placement(transformation(extent={{252,100},{272,120}})));
+  Utilities.IO.SignalExchange.Overwrite oveFan(u(
+      min=0,
+      max=1,
+      unit="1"), description="Integer signal to control the stage of the fan either on or off")
+    "Block for overwriting fan control signal" annotation (Placement(
+        transformation(
+        extent={{10,10},{-10,-10}},
+        rotation=180,
+        origin={202,110})));
+  Utilities.IO.SignalExchange.Read reaFan(description="Control signal for fan",
+      y(unit="1")) "Read control signal for fan"
+    annotation (Placement(transformation(extent={{222,100},{242,120}})));
+  Modelica.Blocks.Sources.RealExpression yFan(y=if heaPum.com.isOn then 1 else 0)
+    "Control input signal to fan"
+    annotation (Placement(transformation(extent={{160,100},{180,120}})));
+  Fluid.Movers.FlowControlled_dp fan(
+    redeclare package Medium = MediumAir,
+    massDynamics=Modelica.Fluid.Types.Dynamics.SteadyState,
+    inputType=IDEAS.Fluid.Types.InputType.Stages,
+    use_inputFilter=false,
+    dp_nominal=100,
+    m_flow_nominal=3,
```

Comment:
3 cubic meters of air per second? That sounds like a lot but it corresponds to 15kW at dT=5 so seems to be realistic.👍